share/wake/lib/system/http.wake

```diff
@@ -21,13 +21,13 @@ tuple HttpHeader =
     Name: String
     Value: String
 
-data HttpFormDataValue =
-    HttpFormDataString String
-    HttpFormDataFile Path
-
+# A form data entry. It consists of a user defined named and a file on disk to upload.
+#
+# Note: while raw strings are generally accepted as http form data the actual encoding of the
+# string restults in shell escape headaches. Strings should be written to a file before upload.
 tuple HttpFormData =
     Name: String
-    Value: HttpFormDataValue
+    File: String
```

Comment:
I know there's cases where using a file would be overkill, but it seems like the curl argument you generate always *expects* it to be a file, so can we use `Path` here instead of `String`?  If you want to accept non-`Path` data to `addFormData`, just have that write the input to a temporary file.

Reply:
It _should_ be a path but I'm fighting with the Runner api for RSC. The returned outputs are returned as a string not a path as they haven't been hashed yet and afaik I can't upgrade them to a Path without either causing 'multiple jobs output file' or breaking something downstream.

as for why its a always a file, that comes down to really long strings messing with your shell/curl so its better to always be a file on disk

Reply:
Ah, yeah, that would be a problem.  Maybe add a comment that they are deliberately left unhashed to avoid the "multiple jobs" error?  Since this is an internal tuple we can enforce that invariant, but only if we know what we're enforcing, and if we don't forget and export this later.

Reply:
Yep SG. In the longer term (I'll leave a todo) it would be good to figure out how to use a path here without breaking rsc